lib/CL/pocl_util.c

```diff
@@ -599,7 +599,7 @@ pocl_create_event_sync (cl_event waiting_event, cl_event notifier_event)
         }
     }
 
-  if (notifier_event->status == CL_COMPLETE)
+  if (notifier_event->status <= CL_COMPLETE)
```

Comment:
This seems creepy. Can you add a comment here?

Reply:
it is to prevent an event sync to be created with an event that has already failed. like completed events, failed events only notify waiting events when the event gets marked as failed. 

ps. this branch is still a wip and some commits might not make it in the final branch

Reply:
OK, the creepiness I meant is due to the all encompassing `<=` operator. Perhaps compare directly to CL_FAILED or whatever the status is for that.

Reply:
ah i see, the thing is that an event can have pretty much any error code, https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#clGetEventInfo

Reply:
Ah, I see. Then I'd even use a magic number here (< 0) and add a comment since the spec says: `Or an error code given by a negative integer value (command was abnormally terminated - this may be caused by a bad memory access etc.). These error codes come from the same set of error codes that are returned from the platform or runtime API calls as return values or errcode_ret values.`

Reply:
I'd say the `<= CL_COMPLETE` statement is understood to mean cl_complete or an error, it is done in more places. I think that the current comment is sufficient in explaining it: 
```
   // If the notifier event is already complete (or failed),
    // don't create an event sync. This is fine since if the wait
    // event has no notifier events and gets submitted, it can start
    // right away.
```

Reply:
Yeah, but the spec says "negative" not "smaller or equal to CL_COMPLETE, whatever that is" :) Not a big deal, just caught my eye.